File: dx/src/uipp/windows/OpenCommand.cpp
/***********************************************************************/
/* Open Visualization Data Explorer                                    */
/* (C) Copyright IBM Corp. 1989,1999                                   */
/* ALL RIGHTS RESERVED                                                 */
/* This code licensed under the                                        */
/*    "IBM PUBLIC LICENSE - Open Visualization Data Explorer"          */
/***********************************************************************/

#include <dxconfig.h>
#include "defines.h"


#include "OpenCommand.h"
#include "DXApplication.h"
#include "DXWindow.h"
#include "Network.h"

OpenCommand::OpenCommand(const char*   name,
                         CommandScope* scope,
                         boolean       active,
			 DXApplication *app) :
    OptionalPreActionCommand(name, scope, active,
                             "Save Confirmation",
                             "Do you want to save the program?")
{
    this->application = app;
}

void   OpenCommand::doPreAction()
{
    DXApplication *app = this->application;
    Network *net = app->network;
    const char    *fname = net->getFileName();

    if(fname)
    {
	if(net->saveNetwork(fname))
	     this->doIt(NULL);
    }
    else {
//	Widget parent = this->dialogParent;
	//if (!parent)
	//    parent = app->getAnchor()->getRootWidget();
 //   	net->postSaveAsDialog(parent, this);
   }

}

boolean OpenCommand::doIt(CommandInterface *ci)
{
   this->application->postOpenNetworkDialog();
   return TRUE;
}

boolean OpenCommand::needsConfirmation()
{
    return this->application->network->saveToFileRequired();
}

File: VisualDX/WinDXui/OpenCommand.cs
//Completed 4/24/2006

using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace WinDX.UI
{
    class OpenCommand : OptionalPreActionCommand
    {
        private DXApplication application;

        public OpenCommand(String name, CommandScope scope, bool active,
            DXApplication app, Form parent)
            : base(name, scope, active,
            "Save Confirmation", "Do you want to save the program?",
            parent)
        {
            this.application = app;
        }

        public OpenCommand(String name, CommandScope scope, bool active,
            DXApplication app)
            : base(name, scope, active,
            "Save Confirmation", "Do you want to save the program?")
        {
            this.application = app;
        }

        protected override bool needsConfirmation()
        {
            return this.application.network.SaveToFileRequired;
        }

        protected override void doPreAction()
        {
            Network net = application.network;
            String fname = net.FileName;

            if (fname != null)
            {
                if (net.saveNetwork(fname))
                {
                    this.doIt(null);
                }
            }
            else
            {
                if (parent != null)
                    net.postSaveAsDialog(parent, this);
                else
                    net.postSaveAsDialog(application.getAnchor(), this);
            }

        }

        public override bool doIt(CommandInterface ci)
        {
            this.application.postOpenNetworkDialog();
            return true;
        }
    }
}
